woocommerce/woocommerce-android
# Edit custom amounts business logic

Closes: #10055 


#### NOTE: Please follow the below PR orders for reviewing
This one!
https://github.com/woocommerce/woocommerce-android/pull/10062

### Description


This PR introduces the capability for users to modify existing custom amounts. This feature ensures users can easily update and make necessary changes to their custom amounts without having to delete and re-enter them.

### Testing instructions

1. Navigate to the `orders` tab
2. Click on "+" button to navigate the the order creation screen
3. Click on the "Add custom amount" button
4. Add a custom amount
5. Custom amount gets added
7. Click on the custom amount row
8. Ensure the amount and the name gets prefilled in the custom amount dialog
9. Make any changes
10. Ensure the changes are reflected in the order creation screen. Ensure the payments section also reflects the new changes.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/4fe60531-e397-4b71-840e-bbb3dd2135e8
